builder/buffer.go

```diff
@@ -115,19 +115,33 @@ func (b *Buffer) WriteEscape(value string) {
 }
 
 func (b Buffer) escape(table, value string) string {
-	if value == "*" {
-		if table == "" {
-			return value
-		}
-		return b.Quoter.ID(table) + ".*"
-	}
-
 	key := escapeCacheKey{table: table, value: value, quoter: b.Quoter}
 	escapedValue, ok := escapeCache.Load(key)
 	if ok {
 		return escapedValue.(string)
 	}
 
+	var escaped_table string
+	if table != "" {
+		if strings.IndexByte(table, '.') >= 0 {
+			parts := strings.Split(table, ".")
+			for i, part := range parts {
+				part = strings.TrimSpace(part)
+				parts[i] = b.Quoter.ID(part)
+			}
+			escaped_table = strings.Join(parts, ".")
+		} else {
+			escaped_table = b.Quoter.ID(table)
+		}
+	}
+
+	if value == "*" {
```

Comment:
no need to cache this I think

we can just do `if table == "" && value == "*" { return value }` in the top most part

Reply:
Done)